notification/slack/channel.go

```diff
@@ -98,8 +101,18 @@ func (s *ChannelSender) Channel(ctx context.Context, channelID string) (*Channel
 }
 
 func (s *ChannelSender) loadChannel(ctx context.Context, channelID string) (*Channel, error) {
+	var err error
 	cfg := config.FromContext(ctx)
 
+	teamID, ok := s.teamIDCache.Get(cfg.Slack.AccessToken)
+	if !ok {
+		teamID, err = GetTeamID(ctx)
+		if err != nil {
+			return nil, err
+		}
+		s.teamIDCache.Add(cfg.Slack.AccessToken, teamID)
+	}
```

Comment:
We should handle this in an atomic way so we don't potentially spam requests instead of one

Reply:
This method can be called multiple times by concurrent HTTP requests, for multiple `channelID`s, all of which will have `ok` being false at first, then all calling `GetTeamID` at the same time.